pkg/kp/rcstore/rc.go

```diff
@@ -0,0 +1,95 @@
+package rcstore
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+)
+
+type ReplicationControllerDesires struct {
+	replicasDesired int
+	disabled        bool
+}
+
+type ReplicationControllerData struct {
+	// It is expected that these fields will never be changed after an RC is created.
+	id           string
+	manifest     *pods.Manifest
+	nodeSelector labels.Selector
+	podLabels    labels.Set
+
+	watcher desireWatcher
+
+	// It is expected that these fields can be changed.
+	ReplicationControllerDesires
```

Comment:
I honestly don't even think we know enough about the design yet to want to split out fields that could change from those that can't. I'm not 100% convinced that, say, `nodeSelector` is immutable. I don't think it makes sense to couple that requirement to its layout in Consul, since that implies the need for data migrations if they do change.

I'd be more comfortable making fewer assumptions at this stage.